sympy/printing/maple.py

```diff
@@ -0,0 +1,305 @@
+                             "expression may not evaluate to anything under "
+                             "some condition.")
+        _coup_list = [
+            ("{c}, {e}".format(c=self._print(c),
+                               e=self._print(e)) if c is not True and c is not S.BooleanTrue else "{e}".format(
+                e=self._print(e)))
+            for e, c in expr.args]
+        _inbrace = ', '.join(_coup_list)
+        return 'piecewise({_inbrace})'.format(_inbrace=_inbrace)
+
+    def _print_Rational(self, expr):
+        p, q = int(expr.p), int(expr.q)
+        return "{p}/{q}".format(p=str(p), q=str(q))
+
+    def _print_Relational(self, expr):
+        lhs_code = self._print(expr.lhs)
+        rhs_code = self._print(expr.rhs)
+        op = expr.rel_op
+        if op in spec_relational_ops:
+            op = spec_relational_ops[op]
+        return "{lhs} {rel_op} {rhs}".format(lhs=lhs_code, rel_op=op, rhs=rhs_code)
+
+    def _print_NumberSymbol(self, expr):
+        return number_symbols[expr]
+
+    def _print_NegativeInfinity(self, expr):
+        return '-infinity'
+
+    def _print_Infinity(self, expr):
+        return 'infinity'
+
+    def _print_Idx(self, expr):
+        return self._print(expr.label)
+
+    def _print_BooleanTrue(self, expr):
+        return "true"
+
+    def _print_BooleanFalse(self, expr):
+        return "false"
+
+    def _print_bool(self, expr):
+        return 'true' if expr else 'false'
+
+    def _print_NaN(self, expr):
+        return 'undefined'
+
+    def _get_matrix(self, expr, sparse=False):
+        if expr.cols == 0 or expr.rows == 0:
+            _strM = 'Matrix([], storage = {storage})'.format(
+                storage='sparse' if sparse else 'rectangular')
+        else:
+            _strM = 'Matrix({list}, storage = {storage})'.format(
+                list=self._print(expr.tolist()),
+                storage='sparse' if sparse else 'rectangular')
+        return _strM
+
+    def _print_MatrixElement(self, expr):
+        return "{parent}[{i_maple}, {j_maple}]".format(
+            parent=self.parenthesize(expr.parent, PRECEDENCE["Atom"], strict=True),
+            i_maple=self._print(expr.i + 1),
+            j_maple=self._print(expr.j + 1))
+
+    def _print_MatrixBase(self, expr):
+        return self._get_matrix(expr, sparse=False)
+
+    def _print_SparseMatrix(self, expr):
+        return self._get_matrix(expr, sparse=True)
+
+    _print_Matrix = \
+        _print_DenseMatrix = \
+        _print_MutableDenseMatrix = \
+        _print_ImmutableMatrix = \
+        _print_ImmutableDenseMatrix = \
+        _print_MatrixBase
+    _print_MutableSparseMatrix = \
+        _print_ImmutableSparseMatrix = \
+        _print_SparseMatrix
+
+    def _print_Identity(self, expr):
+        if isinstance(expr.rows, Integer) or isinstance(expr.rows, IntegerConstant):
+            return self._print(sympy.SparseMatrix(expr))
+        # TODO: Identity Matrix Symbol in Maple.
```

Comment:
I guess it can be printed like
`Matrix(3,shape=identity)`
from https://www.maplesoft.com/support/help/Maple/view.aspx?path=Matrix

Reply:
@sylee957 , Oh, you didn't get my problem. Currently, maple printer have no problem with Identity Matrix with certain size. 

However, when the size of identity matrix is undetermined, there is some problem. I don't know if this feature is supported in Maple.

Reply:
You can use is_number or is_integer to check if the matrix dimensions are numeric.

Reply:
@sylee957 , As the code snippet you extracted. I have already use `if isinstance(expr.rows, Integer) or isinstance(expr.rows, IntegerConstant):` to judge the situation. However, I don't know the what is the syntax of undeter-sized identity matrix in Maple.

Reply:
I personally don't think that there is any problem of printing out dimensions like `Matrix(n, shape=identity)` even though the symbolic matrix dimensions are not supported in other CAS.

Technically, symbols would also mean as 'variable names', which is a feature in every other programming languages, and this is why C or Fortran code printers can print out symbolic expressions.

I would expect this testing case to be printed out as
https://github.com/sympy/sympy/blob/3d9edc991b3c305a5af78cd7ef9d971aa8507820/sympy/printing/tests/test_maple.py#L216-L218
```
"A*(3*Matrix(n, shape=identity) + B)"
```

Reply:
@sylee957 , OK no problem.